Firmware/ultralcd.cpp

```diff
@@ -6929,29 +6941,61 @@ bool lcd_selftest()
 
 	_progress = lcd_selftest_screen(testScreen::extruderFan, _progress, 3, true, 2000);
 #if (defined(FANCHECK) && defined(TACH_0))
-	_result = lcd_selftest_fan_dialog(0);
+	switch (lcd_selftest_fan_auto(0)){		// check extruder Fan
```

Comment:
Bunch of compiler warnings are generated. You can suppress them by adding  default: or by using enum class having only three possible values as return value of this function.

sketch/ultralcd.cpp: In function 'bool lcd_selftest()':
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'heater' not handled in switch [-Wswitch]
  switch (lcd_selftest_fan_auto(0)){  // check extruder Fan
         ^
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'bed' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'endstops' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'motor' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'endstop' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'printFan' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'pulley' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'axis' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'wiringFsensor' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6944:9: warning: enumeration value 'triggeringFsensor' not handled in switch [-Wswitch]